src/Workspaces/Core/Portable/Execution/AbstractAssetProvider.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    /// <summary>
+    /// Provides corresponding data of the given checksum
+    /// </summary>
+    internal abstract class AbstractAssetProvider
```

Comment:
why would we have multiple subclasses of this?

Reply:
The other one is for testing purposes.